Civi/Api4/Generic/Traits/DAOActionTrait.php

```diff
@@ -151,31 +149,6 @@ protected function writeObjects($items) {
     return $result;
   }
 
-  /**
-   * Overrideable function to save items using the appropriate BAO function
-   *
-   * @param array[] $items
-   *   Items already formatted by self::writeObjects
-   * @return \CRM_Core_DAO[]
-   *   Array of saved DAO records
-   */
-  protected function write(array $items) {
-    $saved = [];
-    $baoName = $this->getBaoName();
-
-    $method = method_exists($baoName, 'create') ? 'create' : (method_exists($baoName, 'add') ? 'add' : NULL);
-    // Use BAO create or add method if not deprecated
-    if ($method && !ReflectionUtils::isMethodDeprecated($baoName, $method)) {
-      foreach ($items as $item) {
-        $saved[] = $baoName::$method($item);
-      }
-    }
-    else {
-      $saved = $baoName::writeRecords($items);
```

Comment:
@monishdeb instead of deleting this function I think we just need to change this line to use EntityRepository. We still need to call all the non-deprecated `BAO::create` and `add` functions because they haven't yet been ported to use writeRecord & hooks.